packages/shared/lib/services/sync/orchestrator.service.ts

```diff
@@ -200,30 +215,45 @@ export class Orchestrator {
                 response: connection
             } = await connectionService.getConnection(connectionId as string, providerConfigKey as string, environmentId);
 
-            if (!success) {
+            if (!success || !connection) {
                 return { success: false, error, response: false };
             }
 
             let syncs = syncNames;
 
             if (syncs.length === 0) {
-                syncs = await getSyncNamesByConnectionId(connection?.id as number);
+                syncs = await getSyncNamesByConnectionId(connection.id as number);
             }
 
             for (const syncName of syncs) {
-                const sync = await getSyncByIdAndName(connection?.id as number, syncName);
+                const sync = await getSyncByIdAndName(connection.id as number, syncName);
                 if (!sync) {
+                    throw new Error(`Sync "${syncName}" doest not exists.`);
+                }
+                const schedule = await getSchedule(sync.id as string);
+                if (!schedule) {
                     continue;
                 }
-                const schedule = await getSchedule(sync?.id as string);
 
-                const syncClient = await SyncClient.getInstance();
-                await syncClient?.runSyncCommand(schedule?.schedule_id as string, sync?.id as string, command, activityLogId as number, environmentId);
-                // if they're triggering a sync that shouldn't change the schedule status
-                if (command !== SyncCommand.RUN) {
-                    await updateScheduleStatus(schedule?.schedule_id as string, command, activityLogId as number, environmentId);
+                if (fullResync) {
+                    // We cancel any currently running sync to avoid concurrency issues
+                    const pause = await syncClient.runSyncCommand(schedule.schedule_id, sync.id as string, SyncCommand.CANCEL, activityLogId, environmentId);
+                    if (isErr(pause)) {
+                        continue;
+                    }
+
+                    await clearLastSyncDate(sync.id!);
```

Comment:
Seems like this logic can be abstracted into another function within the orchestrator service

Reply:
Changed ed3472d